### PR TITLE
feat: migrate carte to /app and improve layout

### PR DESCRIPTION
**Pull request by XavierJp**
- solve CSS issues with search filter
- solve header issues (missing searchterm, session etc.)
- migration from /rechercher to /app router
- improve etablissement pointers

What is out of scope :
- add hover behaviour on map markers and search results

### CODE REVIEW

File: app/(header-search)/rechercher/page.tsx [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+  const searchFilterParamsJSON = searchFilterParams.toJSON();
+
+  return (
+    <>
+      <NPSBanner />
+      <HeaderWithAdvancedSearch
+        useSearchBar={true}
+        useAgentCTA={true}
+        useMap={false}
+        searchParams={searchFilterParamsJSON}
```
Comment (johangirod) [inline app/(header-search)/rechercher/page.tsx:48]: On a du JSON d'un côté et du `IParams` attendu de l'autre, mais pas d'erreur typescript ?
Ah, je viens de comprendre. La méthode toJSON est très mal nommé, il s'agit d'avantage d'un `toSerializableObject`

File: components/dirigeants-section/is-personne-morale.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
+import { IEtatCivil, IPersonneMorale } from '#models/immatriculation';
+
+/**
+ * Weird bug happennig here. Webpack build fail when this function is in model/dirigeants.ts
+ * @param toBeDetermined
+ * @returns
+ */
```
Comment (johangirod) [inline components/dirigeants-section/is-personne-morale.ts:7]: Might be because of a circular dependancy.

**Review by johangirod — COMMENTED** [review]
This is a good thing to use proper import of maplibre-gl library inside react instead of using an inserted script. 

However, it appears that there is still a lot of error due to this map logic : 
https://errors.data.gouv.fr/organizations/sentry/issues/?groupStatsPeriod=auto&page=0&query=maplibre&referrer=issue-list&statsPeriod=14d

We might want to fix thoses once and for all at some point.

File: app/(header-search)/rechercher/page.tsx (hunk 1 above)
Comment (XavierJp) [reply app/(header-search)/rechercher/page.tsx:48]: Yes, bien vu

**Pull request merged**